frontend/src/metabase/query_builder/components/QueryVisualization.jsx

```diff
@@ -98,22 +83,30 @@ export const VisualizationEmptyState = ({ className }) => (
   </div>
 );
 
-export const VisualizationRunningState = ({
-  className = "",
-  loadingMessage,
-}) => (
-  <div
-    className={cx(
-      className,
-      "Loading flex flex-column layout-centered text-brand",
-    )}
-  >
-    <LoadingSpinner />
-    <h2 className="Loading-message text-brand text-uppercase my3">
-      {loadingMessage}
-    </h2>
-  </div>
-);
+export function VisualizationRunningState({ className = "", loadingMessage }) {
+  const [isSlow] = useTimeout(SLOW_MESSAGE_TIMEOUT);
+
+  // show the slower loading message only when the loadingMessage is
+  // not customised
+  const message =
+    loadingMessage === "Doing science..." && isSlow()
```

Comment:
Few things:
- The message should be translated
- This logic is very easy to break

Wdyt if we do the following:
- Change `getLoadingMessage()` to `getLoadingMessage(isSlow: boolean)`
- Apply the changes in the default plugin
- Apply the changes in the EE plugin. If the setting value (enum value) is still `doing-science`, only then change the message
- Remove `loadingMessage: getWhiteLabeledLoadingMessage(state)` from the QueryBuilder component. Where needed, use the selector directly, e.g. with `useSelector`.

Reply:
Ok, I've implemented you proposal, please have another look.